build.gradle

```diff
@@ -44,34 +45,45 @@ node {
   workDir = file("${project.buildDir}/node")
   nodeModulesDir = file("${project.projectDir}")
 }
- 
-task dist(type: NpmTask) {
-  args = ['run', 'build', '--prod']
-}
- 
-dist.dependsOn(npm_install)
 
-task createDockerfile(type: Dockerfile, dependsOn: 'dist'){
+task createDockerfile(type: Dockerfile) {
     destFile = project.file('docker/Dockerfile')
    
-    from 'nginx:alpine'
-    copyFile("nginx/nginx.conf.template", "/etc/nginx/nginx.conf.template")
-    workingDir("/usr/share/nginx/html")
-    addFile("dist/mongoose-console", ".")
-    instruction "CMD /bin/sh -c \"envsubst '\$MONGOOSE_HOST' < /etc/nginx/nginx.conf.template > /etc/nginx/nginx.conf.template && exec nginx -g 'daemon off;'\""
+   // NOTE: We're using multistage building to separate containers with code and with production angular-generated files. 
+  
+    // NOTE: Stage 1 - building the angular application (Mongoose Console) 
+    from 'node:8.11.2-alpine as builder'
+    workingDir("/usr/src/app")
+    copyFile("package.json", "./")
+    instruction "RUN npm install"
+    copyFile(".", ".")
+    instruction "RUN npm run build"
+
+    // NOTE: Stage 2 - serving the app via Node.js server 
+    from 'node:alpine'
+    // NOTE: Copying the angular-generated files from builder image
+    instruction "COPY --from=builder /usr/src/app/dist/mongoose-console ."
+
+    // NOTE: Serving the app via Node.js server
+    instruction "COPY --from=builder /usr/src/app/server/node-server.js ."
+    instruction "RUN npm install express"
+    instruction "EXPOSE ${CONSOLE_PORT}"
+    instruction 'CMD [ "node", "node-server.js" ]' 
 }
 
-task buildImage(type: DockerBuildImage, dependsOn: 'createDockerfile'){
+task buildImage(type: DockerBuildImage, dependsOn: 'createDockerfile') {
     inputDir = project.file(".")
     dockerFile = createDockerfile.destFile
     imageId = 'emcmongoose/mongoose-console'
-    tags = ['emcmongoose/mongoose-console:$CONSOLE_VER', 'emcmongoose/mongoose-console:latest']
+    // TODO: Fix hard-coded "0.1.0" image tag. It's implicitly set because gradle didn't recognize ...
+    // ... the environment variable.
+    tags = ['emcmongoose/mongoose-console:0.1.0', 'emcmongoose/mongoose-console:latest']
```

Comment:
Variables listen within .env file are available in docker-compose.yml and inside images and it has been listen with "env_file" property. Thus, they can't be accessed in this place (since it's a Groovy script, as far as I understand). 
Consider use any other method to get image tag from some kind of property (environment/ constant / etc).

Reply:
Just get version from the .env file if it is defined there

Reply:
I've tried it in many ways, but I always ended up having "Broken Pipe" error of the Cradle plugin we're using for building the image. Example: If I'd try to fetch the version tag encapsulated to CONSOLE_VER like this:
`tags = ['emcmongoose/mongoose-console:$CONSOLE_VER', 'emcmongoose/mongoose-console:latest']`

I'll have the following error:

 
> Task :buildImage
Building image using context '/Users/Andrey/Developer/Mongoose/console'.
Using Dockerfile '/Users/Andrey/Developer/Mongoose/console/docker/Dockerfile'
Using tags 'emcmongoose/mongoose-console:$CONSOLE_VER', 'emcmongoose/mongoose-console:latest' for image.

Error during callback
com.github.dockerjava.shaded.org.apache.http.client.ClientProtocolException
        at com.github.dockerjava.shaded.org.apache.http.impl.client.InternalHttpClient.doExecute(InternalHttpClient.java:186)
        at com.github.dockerjava.shaded.org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:71)
        at com.github.dockerjava.shaded.org.glassfish.jersey.apache.connector.ApacheConnector.apply(ApacheConnector.java:450)
        at com.github.dockerjava.shaded.org.glassfish.jersey.client.ClientRuntime.invoke(ClientRuntime.java:278)
        at com.github.dockerjava.shaded.org.glassfish.jersey.client.JerseyInvocation.lambda$invoke$1(JerseyInvocation.java:767)
        at com.github.dockerjava.shaded.org.glassfish.jersey.internal.Errors.process(Errors.java:316)
        at com.github.dockerjava.shaded.org.glassfish.jersey.internal.Errors.process(Errors.java:298)
        at com.github.dockerjava.shaded.org.glassfish.jersey.internal.Errors.process(Errors.java:229)
        at com.github.dockerjava.shaded.org.glassfish.jersey.process.internal.RequestScope.runInScope(RequestScope.java:414)
        at com.github.dockerjava.shaded.org.glassfish.jersey.client.JerseyInvocation.invoke(JerseyInvocation.java:765)
        at com.github.dockerjava.shaded.org.glassfish.jersey.client.JerseyInvocation$Builder.method(JerseyInvocation.java:456)
        at com.github.dockerjava.shaded.org.glassfish.jersey.client.JerseyInvocation$Builder.post(JerseyInvocation.java:357)
        at com.github.dockerjava.jaxrs.async.POSTCallbackNotifier.response(POSTCallbackNotifier.java:29)
        at com.github.dockerjava.jaxrs.async.AbstractCallbackNotifier.call(AbstractCallbackNotifier.java:50)
        at com.github.dockerjava.jaxrs.async.AbstractCallbackNotifier.call(AbstractCallbackNotifier.java:24)
Caused by: com.github.dockerjava.shaded.org.apache.http.client.NonRepeatableRequestException: Cannot retry request with a non-repeatable request entity
        at com.github.dockerjava.shaded.org.apache.http.impl.execchain.RetryExec.execute(RetryExec.java:107)
        at com.github.dockerjava.shaded.org.apache.http.impl.execchain.RedirectExec.execute(RedirectExec.java:110)
        at com.github.dockerjava.shaded.org.apache.http.impl.client.InternalHttpClient.doExecute(InternalHttpClient.java:184)
        ... 14 more
Caused by: java.io.IOException: Broken pipe at Socket[addr=0.0.0.0/0.0.0.0,port=0,localport=0][fd=java.io.FileDescriptor@7b3d4a78; file=/Users/Andrey/Library/Containers/com.docker.docker/Data/docker.sock; connected=true; bound=false]
        at org.newsclub.net.unix.AFUNIXSocketImpl$AFUNIXOutputStream.write(AFUNIXSocketImpl.java:260)
        at com.github.dockerjava.shaded.org.apache.http.impl.io.SessionOutputBufferImpl.streamWrite(SessionOutputBufferImpl.java:124)
        at com.github.dockerjava.shaded.org.apache.http.impl.io.SessionOutputBufferImpl.flushBuffer(SessionOutputBufferImpl.java:136)
        at com.github.dockerjava.shaded.org.apache.http.impl.io.SessionOutputBufferImpl.write(SessionOutputBufferImpl.java:167)
        at com.github.dockerjava.shaded.org.apache.http.impl.io.SessionOutputBufferImpl.write(SessionOutputBufferImpl.java:179)
        at com.github.dockerjava.shaded.org.apache.http.impl.io.SessionOutputBufferImpl.writeLine(SessionOutputBufferImpl.java:219)
        at com.github.dockerjava.shaded.org.apache.http.impl.io.ChunkedOutputStream.flushCacheWithAppend(ChunkedOutputStream.java:123)
        at com.github.dockerjava.shaded.org.apache.http.impl.io.ChunkedOutputStream.write(ChunkedOutputStream.java:179)
        at com.github.dockerjava.shaded.org.glassfish.jersey.message.internal.CommittingOutputStream.write(CommittingOutputStream.java:224)
        at com.github.dockerjava.shaded.org.glassfish.jersey.message.internal.WriterInterceptorExecutor$UnCloseableOutputStream.write(WriterInterceptorExecutor.java:300)
        at com.github.dockerjava.shaded.org.glassfish.jersey.message.internal.ReaderWriter.writeTo(ReaderWriter.java:117)
        at com.github.dockerjava.shaded.org.glassfish.jersey.message.internal.AbstractMessageReaderWriterProvider.writeTo(AbstractMessageReaderWriterProvider.java:80)
        at com.github.dockerjava.shaded.org.glassfish.jersey.message.internal.InputStreamProvider.writeTo(InputStreamProvider.java:106)
        at com.github.dockerjava.shaded.org.glassfish.jersey.message.internal.InputStreamProvider.writeTo(InputStreamProvider.java:61)
        at com.github.dockerjava.shaded.org.glassfish.jersey.message.internal.WriterInterceptorExecutor$TerminalWriterInterceptor.invokeWriteTo(WriterInterceptorExecutor.java:266)
        at com.github.dockerjava.shaded.org.glassfish.jersey.message.internal.WriterInterceptorExecutor$TerminalWriterInterceptor.aroundWriteTo(WriterInterceptorExecutor.java:251)
        at com.github.dockerjava.shaded.org.glassfish.jersey.message.internal.WriterInterceptorExecutor.proceed(WriterInterceptorExecutor.java:163)
        at com.github.dockerjava.jaxrs.filter.LoggingFilter.aroundWriteTo(LoggingFilter.java:300)
        at com.github.dockerjava.shaded.org.glassfish.jersey.message.internal.WriterInterceptorExecutor.proceed(WriterInterceptorExecutor.java:163)
        at com.github.dockerjava.shaded.org.glassfish.jersey.message.internal.MessageBodyFactory.writeTo(MessageBodyFactory.java:1135)
        at com.github.dockerjava.shaded.org.glassfish.jersey.client.ClientRequest.doWriteEntity(ClientRequest.java:516)
        at com.github.dockerjava.shaded.org.glassfish.jersey.client.ClientRequest.writeEntity(ClientRequest.java:498)
        at com.github.dockerjava.shaded.org.glassfish.jersey.apache.connector.ApacheConnector$1.writeTo(ApacheConnector.java:598)
        at com.github.dockerjava.shaded.org.apache.http.impl.execchain.RequestEntityProxy.writeTo(RequestEntityProxy.java:123)
        at com.github.dockerjava.shaded.org.apache.http.impl.DefaultBHttpClientConnection.sendRequestEntity(DefaultBHttpClientConnection.java:156)
        at com.github.dockerjava.shaded.org.apache.http.impl.conn.CPoolProxy.sendRequestEntity(CPoolProxy.java:162)
        at com.github.dockerjava.shaded.org.apache.http.protocol.HttpRequestExecutor.doSendRequest(HttpRequestExecutor.java:238)
        at com.github.dockerjava.shaded.org.apache.http.protocol.HttpRequestExecutor.execute(HttpRequestExecutor.java:123)
        at com.github.dockerjava.shaded.org.apache.http.impl.execchain.MainClientExec.execute(MainClientExec.java:271)
        at com.github.dockerjava.shaded.org.apache.http.impl.execchain.ProtocolExec.execute(ProtocolExec.java:184)
        at com.github.dockerjava.shaded.org.apache.http.impl.execchain.RetryExec.execute(RetryExec.java:88)
        ... 16 more
Caused by: org.newsclub.net.unix.AFUNIXSocketException: Broken pipe
        at org.newsclub.net.unix.NativeUnixSocket.write(Native Method)
        at org.newsclub.net.unix.AFUNIXSocketImpl$AFUNIXOutputStream.write(AFUNIXSocketImpl.java:251)
        ... 46 more

> Task :buildImage FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':buildImage'.
> com.github.dockerjava.shaded.org.apache.http.client.ClientProtocolException

Reply:
I don't see any groovy code extracting the version from the .env file